enhancements/assets-generation/cloud-foundry-application-discovery/README.md

```diff
@@ -74,40 +74,233 @@ to implement the design.  For instance,
+worker: bundle exec rake workers:start
+```
+
+In a Procfile, you declare one process type per line and use the syntax
+`PROCESS_TYPE: COMMAND`.
+
+* `PROCESS_TYPE` defines the type of the process.  
+* `COMMAND` is the command line to launch the process.
+
+###### Procfile use cases
+
+Many buildpacks provide their own process types and commands by default; however,
+there are special cases where specifying a custom `COMMAND` is necessary.
+Commands can be overwritten by providing a Procfile with the same process type.
+
+For example, a buildpack may provide a `worker` process type that runs the
+`rake default:start` command. If a Procfile is provided that also contains a
+`worker` process type, but a different command such as `rake custom:start`, the
+`rake custom:start` command will be used.
+
+Some buildpacks, such as Python, that work on a variety of frameworks, do not 
+attempt to provide a default start command. For these cases, a Procfile should
+be used to specify any necessary commands for the app.
+
+###### Web process
+
+`web` is a [special process type](https://v3-apidocs.cloudfoundry.org/version/3.163.0/#web-process-type)
+that is required for all applications. The `web` `PROCESS_TYPE` must be specified
+by either the buildpack or the Procfile.
+
+###### Specifying processes in manifest files
+
+Custom process types can also be configured via a manifest file. Read more about
+[manifests](https://v3-apidocs.cloudfoundry.org/version/3.163.0/#manifests).
+It is not recommended to specify processes in both a manifest and a Procfile for
+the same app.
+
+#### Route-level configuration
+
+This [configuration](https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#routes)
+is for *creating* mappings between the app and a route. Each route is created if
+it does not already exist. The protocol will be updated for any existing route
+mapping.
+
+Example:
+
+```
+---
+  ...
+  routes:
+  - route: example.com
+    protocol: http2
+  - route: www.example.com/foo
+  - route: tcp-example.com:1234
+```
+
+##### Definition
+
+| Name | Type | Description |
+| ----- | ----- | ----- |
+| **route** | *string* | **(Required)** The route URI |
+| **protocol** | *string* | (Optional) Protocol to use for this route. Valid protocols are `http1`, `http2`, and `tcp`. |
+
+#### Service-level configuration
+
+This configuration is *creating* new service bindings between the app and a 
+service instance. The `services` field can take either an array of service 
+instance name strings or an array of the following service-level fields.
+
+##### Definition
+
+| Name | Type | Description |
+| ----- | ----- | ----- |
+| **name** | *string* | **(Required)** The name of the service instance to be bound to |
+| **binding\_name** | *string* | The name of the service binding to be created |
+| **parameters** | *object* | A map of arbitrary key/value pairs to send to the service broker during binding |
+
+#### Sidecar-level configuration
```

Comment:
Do you think it's worth adding a link for each section? Like this one for sidecars?
https://v3-apidocs.cloudfoundry.org/version/3.163.0/#sidecars